File: business/business-oj/src/main/java/com/glowxq/oj/user/pojo/po/UserInfo.java
package com.glowxq.oj.user.pojo.po;

import com.glowxq.core.common.entity.base.BaseEntity;
import com.glowxq.mysql.EntityChangeListener;
import com.mybatisflex.annotation.Id;
import com.mybatisflex.annotation.KeyType;
import com.mybatisflex.annotation.Table;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;

import java.io.Serial;
import java.time.LocalDate;
import java.time.LocalDateTime;

/**
 * <p>
 * 用户信息
 * </p>
 *
 * @author glowxq
 * @since 2025-06-23
 */
@Data
@Table(value = "user_info", onInsert = EntityChangeListener.class, onUpdate = EntityChangeListener.class)
@Schema(description = "用户信息")
public class UserInfo implements BaseEntity {

    @Serial
    private static final long serialVersionUID = 1L;

    @Id(keyType = KeyType.None)
    @Schema(description = "ID")
    private Long id;

    @Schema(description = "用户ID")
    private Long userId;

    @Schema(description = "部门ID")
    private Long deptId;

    @Schema(description = "姓名")
    private String name;

    @Schema(description = "昵称")
    private String nickName;

    @Schema(description = "手机号")
    private String phone;

    @Schema(description = "邮箱")
    private String email;

    @Schema(description = "头像信息")
    private String avatar;

    @Schema(description = "性别")
    private String sex;

    @Schema(description = "生日")
    private LocalDate birthday;

    @Schema(description = "相关图片")
    private String image;

    @Schema(description = "ac数量")
    private Integer acNum;

    @Schema(description = "积分")
    private Integer integral;

    @Schema(description = "连续签到时间")
    private Integer continuousSignDay;

    @Schema(description = "提交题目数量")
    private Integer submitNum;

    @Schema(description = "称号")
    private String title;

    @Schema(description = "颜色")
    private String color;

    @Schema(description = "备注")
    private String remark;

    @Schema(description = "帐号过期时间")
    private LocalDate expirationTime;

    @Schema(description = "最后登录时间")
    private LocalDateTime lastLoginTime;

    @Schema(description = "最后登录ip")
    private String lastLoginIp;

    @Schema(description = "创建时间")
    private LocalDateTime createTime;

    @Schema(description = "更新时间")
    private LocalDateTime updateTime;

    @Schema(description = "是否删除")
    private String delFlag;

    @Schema(description = "创建人ID")
    private Long createId;

    @Schema(description = "更新人ID")
    private Long updateId;

    @Schema(description = "租户ID")
    private String tenantId;

    public void addIntegral(int signAddIntegral) {
        this.integral += signAddIntegral;
    }

    public void plusSubmitNum() {
        this.submitNum += 1;
    }

    public void acNumPlus() {
        this.acNum += 1;
    }
}

File: oj-web/src/api/interface/oj/user/userInfo.ts
import type { IPageQuery } from '@/api/interface'
import type { IRole } from '@/api/interface/system/admin/role'
import type { IMetaTag } from '@/api/interface/system/meta/metaTag'
import type { IGroup } from '@/api/interface/oj/group/group'

export namespace IUserInfo {

  // 查询条件
  export interface Query extends IPageQuery {
    userId?: number
    deptId?: number
    name?: string
    nickName?: string
    phone?: string
    email?: string
    avatar?: string
    sex?: string
    birthdayStart?: string
    birthdayEnd?: string
    image?: string
    acNum?: number
    integral?: number
    continuousSignDay?: number
    submitNum?: number
    title?: string
    color?: string
    expirationTimeStart?: string
    expirationTimeEnd?: string
    lastLoginTimeStart?: string
    lastLoginTimeEnd?: string
    lastLoginIp?: string
    tenantId?: string
  }

  // 编辑form表单
  export interface Form {
    id?: number
    userId?: number
    deptId?: number
    name?: string
    nickName?: string
    phone?: string
    email?: string
    avatar?: string
    sex?: string
    birthday?: string
    image?: string
    acNum?: number
    integral?: number
    continuousSignDay?: number
    submitNum?: number
    title?: string
    color?: string
    remark?: string
    expirationTime?: string
    lastLoginTime?: string
    lastLoginIp?: string
    tenantId?: string
    groupIds?: number[]
    roleIds?: number[]
  }

  // list或detail返回结构
  export interface Row {
    id?: number
    userId?: number
    deptId?: number
    name?: string
    nickName?: string
    phone?: string
    email?: string
    avatar?: string
    sex?: string
    birthday?: string
    image?: string
    acNum?: number
    integral?: number
    continuousSignDay?: number
    submitNum?: number
    title?: string
    color?: string
    remark?: string
    expirationTime?: string
    lastLoginTime?: string
    lastLoginIp?: string
    tenantId?: string
    groupIds?: number[]
    roleIds?: number[]
    createTime?: string
    updateTime?: string
    // 新增字段
    roles?: IRole.Info[]
    tags?: IMetaTag.Row[]
    groups?: IGroup.Row[]
  }

  // OJ用户注册接口
  export interface RegisterForm {
    name: string
    nickName?: string
    phone?: string
    email?: string
    avatar?: string
    sex?: string
    birthday?: string
    groupCode?: string
  }

} 